Src/java/qdm/src/main/resources/gov/healthit/qdm/qdm-modelinfo-4.2.xml

```diff
@@ -0,0 +1,541 @@
+    </ns4:typeInfo>
+    <ns4:typeInfo xsi:type="ns4:ClassInfo" name="QDM.Symptom" baseType="QDM.QDMBaseType">
+        <ns4:element name="onsetDatetime" type="System.DateTime"/>
+        <ns4:element name="abatementDatetime" type="System.DateTime"/>
+        <ns4:element name="severity" type="System.Concept"/>
+    </ns4:typeInfo>
+    <ns4:typeInfo xsi:type="ns4:ClassInfo" name="QDM.MedicationAdministered" baseType="QDM.QDMBaseType">
+        <ns4:element name="startDatetime" type="System.DateTime"/>
+        <ns4:element name="stopDatetime" type="System.DateTime"/>
+        <ns4:element name="negationRationale" type="System.Concept"/>
+        <ns4:element name="dose" type="System.Quantity"/>
+        <ns4:element name="frequency" type="System.Concept"/>
+        <ns4:element name="route" type="System.Concept"/>
+        <ns4:element name="reason" type="System.Concept"/>
+        <ns4:element name="cumaltiveMedicationDuration" type="System.Quantity"/>
+    </ns4:typeInfo>
+    <ns4:typeInfo xsi:type="ns4:ClassInfo" name="QDM.RiskCategoryAssessment" baseType="QDM.QDMBaseType">
+        <ns4:element name="startDatetime" type="System.DateTime"/>
+        <ns4:element name="stopDatetime" type="System.DateTime"/>
+        <ns4:element name="negationRationale" type="System.Concept"/>
+        <ns4:element name="result" type="System.Concept"/>
+    </ns4:typeInfo>
+    <ns4:typeInfo xsi:type="ns4:ClassInfo" name="QDM.ProcedureRecommended" baseType="QDM.QDMBaseType">
+        <ns4:element name="startDatetime" type="System.DateTime"/>
+        <ns4:element name="stopDatetime" type="System.DateTime"/>
+        <ns4:element name="negationRationale" type="System.Concept"/>
+        <ns4:element name="reason" type="System.Concept"/>
+        <ns4:element name="method" type="System.Concept"/>
+        <ns4:element name="anatomicalApproachSite" type="System.Concept"/>
+        <ns4:element name="anatomicalLocationSite" type="System.Concept"/>
+        <ns4:element name="ordinality" type="System.Concept"/>
+    </ns4:typeInfo>
+    <ns4:typeInfo xsi:type="ns4:ClassInfo" name="QDM.EncounterPerformed" baseType="QDM.QDMBaseType"
+                  retrievable="true" label="Encounter, Performed" identifier="EncounterPerformed" primaryCodePath="code">
+        <ns4:element name="admissionDatetime" type="System.DateTime"/>
+        <ns4:element name="dischargeDatetime" type="System.DateTime"/>
+        <ns4:element name="lengthOfStay" type="System.Quantity"/>
+        <ns4:element name="dischargeStatus" type="System.Concept"/>
+        <ns4:element name="facilityLocation" type="System.Concept"/>
+        <ns4:element name="facilityLocationArrivalDatetime" type="System.DateTime"/>
+        <ns4:element name="facilityLocationDepartureDatetime" type="System.DateTime"/>
+        <ns4:element name="reason" type="System.Concept"/>
+        <ns4:element name="diagnosis" type="QDM.Diagnosis"/>
+        <ns4:element name="principalDiagnosis" type="QDM.Diagnosis"/>
```

Comment:
It is true that the QDM-based HQMF used a non-flat representation, but the authoring format (QDM) has always been flat.  My assumption was that the QDM 4.2 based data model would stick to the same basic constraints as the QDM 4.2 authoring language.

If you choose to make the data model non-flat, you should make sure that the MAT and Bonnie teams  are aware and on-board, as they need to develop a UI and implementation that supports it.